sim/hunter/item_sets_pve_phase_8.go

```diff
@@ -0,0 +1,216 @@
+	"github.com/wowsims/sod/sim/core/stats"
+)
+
+var ItemSetDawnstalkerProwess = core.NewItemSet(core.ItemSet{
+	Name: "Dawnstalker Prowess",
+	Bonuses: map[int32]core.ApplyEffect{
+		2: func(agent core.Agent) {
+			hunter := agent.(HunterAgent).GetHunter()
+			hunter.applyScarletEnclaveMelee2PBonus()
+		},
+		4: func(agent core.Agent) {
+			hunter := agent.(HunterAgent).GetHunter()
+			hunter.applyScarletEnclaveMelee4PBonus()
+		},
+		6: func(agent core.Agent) {
+			hunter := agent.(HunterAgent).GetHunter()
+			hunter.applyScarletEnclaveMelee6PBonus()
+		},
+	},
+})
+
+// Your Strikes and Mongoose Bite deal 20% increased damage to targets afflicted with your Serpent Sting or Wyvern Strike.
+func (hunter *Hunter) applyScarletEnclaveMelee2PBonus() {
+	label := "S03 - Item - Scarlet Enclave - Hunter - Melee 2P Bonus"
+	if hunter.HasAura(label) {
+		return
+	}
+
+	damageMod := hunter.AddDynamicMod(core.SpellModConfig{
+		Kind:      core.SpellMod_DamageDone_Flat,
+		ClassMask: ClassSpellMask_HunterStrikes | ClassSpellMask_HunterMongooseBite,
+	})
+
+	core.MakeProcTriggerAura(&hunter.Unit, core.ProcTrigger{
+		Name:           label,
+		ClassSpellMask: ClassSpellMask_HunterStrikes | ClassSpellMask_HunterMongooseBite,
+		Callback:       core.CallbackOnApplyEffects,
+		Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			hasDebuff := result.Target.HasActiveAuraWithTag("SerpentSting") || result.Target.HasActiveAuraWithTag("WyvernStrike - Bleed")
+			damageMod.UpdateIntValue(core.TernaryInt64(hasDebuff, 20, 0))
+			damageMod.Activate()
+		},
+	})
+}
+
+// Your melee critical strikes increase your attack speed by 20% for 10 sec.
+func (hunter *Hunter) applyScarletEnclaveMelee4PBonus() {
+	label := "S03 - Item - Scarlet Enclave - Hunter - Melee 4P Bonus"
+	if hunter.HasAura(label) {
+		return
+	}
+
+	attackSpeed := 1.20
+
+	procAura := core.MakePermanent(hunter.RegisterAura(core.Aura{
+		Label:    "Wicked Fast",
+		ActionID: core.ActionID{SpellID: 1226357},
+		Duration: time.Second * 10,
+		OnGain: func(_ *core.Aura, sim *core.Simulation) {
+			hunter.MultiplyMeleeSpeed(sim, attackSpeed)
+		},
+		OnExpire: func(_ *core.Aura, sim *core.Simulation) {
+			hunter.MultiplyMeleeSpeed(sim, 1/attackSpeed)
+		},
+	}))
+
+	core.MakeProcTriggerAura(&hunter.Unit, core.ProcTrigger{
+		Name:     label,
+		Callback: core.CallbackOnSpellHitDealt,
+		Outcome:  core.OutcomeCrit,
+		ProcMask: core.ProcMaskMelee,
+		Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			procAura.Activate(sim)
+		},
+	})
+}
+
+// Increases the bonus damage from Raptor Fury by an additional 10% per stack.
+func (hunter *Hunter) applyScarletEnclaveMelee6PBonus() {
+	label := "S03 - Item - Scarlet Enclave - Hunter - Melee 6P Bonus"
+	if hunter.HasAura(label) {
+		return
+	}
+
+	if !hunter.HasRune(proto.HunterRune_RuneBracersRaptorFury) {
+		return
+	}
+
+	hunter.BonusRaptorFuryDamageMultiplier = 0.10
+}
```

Comment:
🛑 we should still initialize an aura here (just in case). Even though there's no dual application of these sets right now it's good to keep it consistent